lib/task_005_lib.rb

```diff
@@ -0,0 +1,95 @@
+module Task005
+  module_function
+
+  class Point
+    attr_reader :x, :y
+
+    def self.from_input(point)
+      x, y = point.split(',').map(&:to_i)
+      Point.new(x, y)
+    end
+
+    def initialize(x, y)
+      @x = x
+      @y = y
+    end
+  end
+
+  class Line
+    attr_reader :point1, :point2
+
+    def self.from_input(input)
+      point1, point2 = input.split(' -> ').map { |point| Point.from_input(point) }
+      Line.new(point1, point2)
+    end
+
+    def initialize(point1, point2)
+      @point1 = point1
+      @point2 = point2
+    end
+
+    def horizontal_or_vertical?
+      @point1.x == @point2.x || @point1.y == @point2.y
+    end
+
+    def degree_45?
+      (@point1.x - @point2.x).abs == (@point1.y - @point2.y).abs
+    end
+
+    def points_on_line
+      if horizontal_or_vertical?
+        points = []
+        x1, x2 = [@point1.x, @point2.x].sort
+        y1, y2 = [@point1.y, @point2.y].sort
+        (x1..x2).each do |x|
+          (y1..y2).each do |y|
+            points << [x, y]
+          end
```

Comment:
Would you be able to avoid using temporary helper variable (`points`)?

It should be possible here and in the other case (degree 45) as well :) I can write a sample solution for the `horizontal_or_vertical` if you want me to share it with you. Just let me know.